### PR TITLE
Fix issue with config decl at class level

### LINKED ISSUE #2522: `BaseSettings`-specific configuration cannot be passed to class as keyword arguments
### Checks

* [x] I added a descriptive title to this issue
* [x] I have searched (google, github) for similar issues and couldn't find anything
* [x] I have read and followed [the docs](https://pydantic-docs.helpmanual.io/) and still think this is a bug

# Bug

Output of `python -c "import pydantic.utils; print(pydantic.utils.version_info())"`:
```
             pydantic version: 1.8.1
            pydantic compiled: False
                 install path: [...]/lib/python3.9/site-packages/pydantic
               python version: 3.9.2 (default, Feb 21 2021, 06:38:26)  [Clang 7.1.0 (tags/RELEASE_710/final)]
                     platform: macOS-10.14.6-x86_64-i386-64bit
     optional deps. installed: ['typing-extensions']
```

The `BaseModel` metaclass ignores configuration settings specific to `BaseSettings` when these are passed as keyword arguments during class creation.  They are passed on to `ABCMeta` which does not accept keyword arguments:

```py
import pydantic

class Foo(pydantic.BaseSettings, env_prefix='BAR_'):
    pass

```
Output:

```
Traceback (most recent call last):
  File "[...]/foo.py", line 3, in <module>
    class Foo(pydantic.BaseSettings, env_prefix='BAR_'):
  File "[...]/lib/python3.9/site-packages/pydantic/main.py", line 361, in __new__
    cls = super().__new__(mcs, name, bases, new_namespace, **kwargs)
  File "/nix/store/cdifcrgj5phbvspzs02yglz35qql4i1x-python3-3.9.2/lib/python3.9/abc.py", line 85, in __new__
    cls = super().__new__(mcls, name, bases, namespace, **kwargs)
TypeError: Foo.__init_subclass__() takes no keyword arguments
```

### PR DESCRIPTION
**Pull request by uriyyo**
## Change Summary
Fix issue with config decl at class level

## Related issue number
Fixes #2522

## Checklist

* [x] Unit tests for the changes exist
* [x] Tests pass on CI and coverage remains at 100%
* [x] Documentation reflects the changes where applicable
* [x] `changes/<pull request or issue id>-<github username>.md` file added describing change
  (see [changes/README.md](https://github.com/samuelcolvin/pydantic/blob/master/changes/README.md) for details)

### CODE REVIEW

File: pydantic/main.py [hunk 1]
```diff
@@ -248,7 +248,12 @@ def __new__(mcs, name, bases, namespace, **kwargs):  # noqa C901
                 class_vars.update(base.__class_vars__)
                 hash_func = base.__hash__
 
-        config_kwargs = {key: kwargs.pop(key) for key in kwargs.keys() & BaseConfig.__dict__.keys()}
+        allowed_configs: SetStr = {
+            key
+            for key in dir(config)
+            if not (key.startswith('__') and key.endswith('__'))  # skip dander methods and attributes
+        }
+        config_kwargs = {key: kwargs.pop(key) for key in kwargs.keys() & allowed_configs}
```
Comment (PrettyWood) [inline pydantic/main.py:256]:
```suggestion
        allowed_config_kwargs: SetStr = {
            key
            for key in dir(config)
            if not (key.startswith('__') and key.endswith('__'))  # skip dunder methods and attributes
        }
        config_kwargs = {key: kwargs.pop(key) for key in kwargs.keys() & allowed_config_kwargs}
```

File: changes/2532-uriyyo.md [hunk 2]
```diff
@@ -0,0 +1 @@
+Fix issue with config decl at class level.
```
Comment (PrettyWood) [inline changes/2532-uriyyo.md:1]: Do you think you could improve the text to make it more explicit please?

**Review by PrettyWood — COMMENTED** [review]
LGTM otherwise thanks!

File: changes/2532-uriyyo.md (hunk 2 above)
Comment (uriyyo) [reply changes/2532-uriyyo.md]: Done

**Review by PrettyWood — APPROVED** [review]
Thanks!

**Comment by samuelcolvin** [comment]
great, thank you so much.

**Pull request merged**